Hunting Queries/SecurityEvent/Least_Common_Parent_Child_Process.txt

```diff
@@ -0,0 +1,19 @@
+// Name: Least Common Parent And Child Process Pairs
+// Id: 3712595d-6f47-416b-963a-605201ed2764
+// Description: Looks across your environment for least common Parent/Child process combinations.  Will possibly find some malicious activity disguised as well known process names.  By ZanCo
+// DataSource: #SecurityEvent
+// Tactics: #InitialAccess, #Execution, #Persistance
+let Whitelist = dynamic (['foo.exe', 'baz.exe']);
+let Sensitivity = 5;
+let StartDate = ago(7d);
+let Duration = 7d;
+SecurityEvent
+| where EventID == 4688 and TimeGenerated > StartDate and TimeGenerated < (StartDate + Duration)
+| extend ProcArray = split(NewProcessName, '\\'), ParentProcArray = split(ParentProcessName, '\\')
+| extend ProcArrayLength = arraylength(ProcArray), ParentProcArrayLength = arraylength(ParentProcArray)
+| extend LastIndex = ProcArrayLength - 1, ParentLastIndex = ParentProcArrayLength - 1
+| extend Proc = ProcArray[LastIndex], ParentProc = ParentProcArray[ParentLastIndex]
+| where Proc !in (Whitelist)
+| extend ParentChildPair = strcat(ParentProc , ' > ', Proc)
+| summarize TimesSeen = count(), Hosts = makeset(Computer), Users = makeset(SubjectUserName) by ParentChildPair
```

Comment:
Seems like adding dcount for number of hosts and users would be useful in later pivoting:
| summarize TimesSeen = count(), HostCount = dcount(Computer), Hosts = makeset(Computer), UserCount = dcount(SubjectUserName), Users = makeset(SubjectUserName) by ParentChildPair

Reply:
added to all applicable queries